imcsdk/imcsession.py

```diff
@@ -415,36 +415,41 @@ def __validate_connection(self):
 
     def __validate_imc(self):
         """ ValidateIMC method validates if a given host is valid IMC Server."""
-        from .imcmethodfactory import config_resolve_class
+
+        from .imcmethodfactory import config_resolve_class, config_resolve_dn
+        is_imc = False
+
+        nw_elem = config_resolve_class(cookie=self.__cookie,
+                                       class_id="networkElement")
+        try:
+            nw_elem_response = self.post_elem(nw_elem)
+            if nw_elem_response.error_code == 0:
+                self._logout()
+            else:
+                is_imc = True
+        except:
+            is_imc = True
+        if not is_imc:
+            return False
+
         valid_models = ("R460-4640810", "C260-BASE-2646")
         model_validated = False
 
-        rack_elem = config_resolve_class(cookie=self.__cookie,
-                                         class_id="computeRackUnit")
-
+        rack_elem = config_resolve_dn(cookie=self.__cookie, dn="sys/rack-unit-1")
```

Comment:
Why are we using hardcoded dn? Will this dn `sys/rack-unit-1` be always present?